pinot-server/src/main/java/org/apache/pinot/server/starter/helix/FreshnessBasedConsumptionStatusChecker.java

```diff
@@ -76,15 +90,36 @@ protected boolean isSegmentCaughtUp(String segmentName, LLRealtimeSegmentDataMan
     StreamPartitionMsgOffset currentOffset = rtSegmentDataManager.getCurrentOffset();
     StreamPartitionMsgOffset latestStreamOffset = rtSegmentDataManager.fetchLatestStreamOffset(5000);
     if (isOffsetCaughtUp(currentOffset, latestStreamOffset)) {
-      _logger.info("Segment {} with freshness {}ms has not caught up within min freshness {}."
+      _logger.info("Segment {} with freshness {}ms has not caught up within min freshness {}. "
               + "But the current ingested offset is equal to the latest available offset {}.", segmentName, freshnessMs,
           _minFreshnessMs, currentOffset);
       return true;
     }
 
+    // If the earliest available kafka offset is later than the current offset, then for the purposes of
+    // this check, we are caught up. This can happen if the segment has not consumed data for some time,
+    // and that data has since been retentioned.
+    StreamPartitionMsgOffset earliestStreamOffset = rtSegmentDataManager.fetchEarliestStreamOffset(5000);
```

Comment:
Do we want to count this as pass? If the consumer works as expected, it should be able to skip to the next available offset